lib/match.rb

```diff
@@ -0,0 +1,147 @@
+require_relative "board"
+require_relative "player"
+
+class Match
```

Comment:
Our `Match` class appears to be doing too much work here. I can hear him complaining to the labor unions. :D Ok pardon my corny joke. 

Let's refactor our `Match`. I recommend a template that you can use for this:

```ruby
class Match
  def initialize(board, player1, player2)
    # our match probably only needs to have 3 instance variables
   # @player1, @player2 and @board
   end

   def game_loop
     until game_is_over?
      # 1. @player1 makes a turn unless game_is_over?
     # 2. @player2 makes a turn unless game_is_over?
     end
     # call needed method for when the game_is_over?
   end

   private
  
   def game_is_over?
    # this method simply checks for the ff:
    # has @player1 won? || has @player2 won? || is @board already full?
   end
end
```

Reply:
Our game_is_over works with the evulation winner? || tie? . If player 1 has won then it means that player2 has lost, so there is no need to check if player2 has won. So the way we implemented it was that every time a player makes a move, we check if the current token turn fills the condition for winning, so if that happens, our winner? method tells that the last player that played has won. I think that simplifies the idea of evaluating two players for just one. And that's re-engineering right? 
Our tie? method checks if the board is full, so it actually works the same. The thing is that we store the moves in the match class because the responsability of the match is to take control of the turns, so that's way.